shelley/chain-and-ledger/executable-spec/test/Rules/TestPool.hs

```diff
@@ -102,3 +107,40 @@ poolRetireInEpoch = withTests (fromIntegral numberOfTests) . property $ do
                                     && Maybe.maybe False ((== e) . epochFromSlot) (stp' !? certWit))
             _                  -> False
         registeredPoolRetired _ _ _ = True
+
+-- | Check that a `RegPool` certificate properly adds a stake pool.
+registeredPoolIsAdded
+  :: MonadTest m
+  => Environment LEDGER
+  -> [SourceSignalTarget POOL]
+  -> m ()
+registeredPoolIsAdded env ssts =
+  assertAll addedRegPool ssts
+
+ where
+
+  addedRegPool :: SourceSignalTarget POOL
+               -> Bool
+  addedRegPool sst =
+    case signal sst of
+      RegPool poolParams -> check poolParams
+      _ -> True
+   where
+    check :: PoolParams -> Bool
+    check poolParams =
+      let hk = poolParams ^. poolPubKey
+          pSt = target sst
+          -- PoolParams are registered in pParams map
+       in M.lookup hk (pSt ^. pParams) == Just poolParams
+          -- Hashkey is registered in stPools map
+       && M.lookup hk (pSt ^. stPools . to (\(StakePools x) -> x))
+            == Just (ledgerSlot env)
+          -- Hashkey is registered in cCounters map
+       && hk ∈ M.keys (pSt ^. cCounters)
+
+-- | Transform LEDGER `SourceSignalTargets`s to POOL ones.
```

Comment:
this function could move up a level to TestLedger, so that TestPool remains dumb about the LEDGER STS (and only references the POOL STS)

Reply:
Yeah, agree it makes sense to contain the LEDGER logic in TestLedger. Just pushed a changeset.